metadata-ingestion/src/datahub/ingestion/source/aws/path_spec.py

```diff
@@ -96,28 +88,83 @@ def get_parsable_include(cls, include: str) -> str:
+    @pydantic.validator("file_types", always=True)
+    def validate_file_types(cls, v: Optional[List[str]]) -> List[str]:
+        if v is None:
+            return SUPPORTED_FILE_TYPES
         else:
-            for file_type in values["file_types"]:
+            for file_type in v:
                 if file_type not in SUPPORTED_FILE_TYPES:
                     raise ValueError(
                         f"file type {file_type} not in supported file types. Please specify one from {SUPPORTED_FILE_TYPES}"
                     )
+            return v
 
-        if values.get("default_extension") is not None:
-            if values.get("default_extension") not in SUPPORTED_FILE_TYPES:
-                raise ValueError(
-                    f"default extension {values.get('default_extension')} not in supported default file extension. Please specify one from {SUPPORTED_FILE_TYPES}"
+    @pydantic.validator("default_extension")
+    def validate_default_extension(cls, v):
+        if v not in SUPPORTED_FILE_TYPES:
+            raise ValueError(
+                f"default extension {v} not in supported default file extension. Please specify one from {SUPPORTED_FILE_TYPES}"
+            )
+        return v
+
+    @pydantic.validator("sample_files", always=True)
+    def turn_off_sampling_for_non_s3(cls, v, values):
+        is_s3 = is_s3_uri(values.get("include") or "")
+        if not is_s3:
+            # Sampling only makes sense on s3 currently
+            v = False
+        return v
+
+    @pydantic.validator("exclude", each_item=True)
+    def no_named_fields_in_exclude(cls, v: str) -> str:
+        if len(parse.compile(v).named_fields) != 0:
+            raise ValueError(
+                f"path_spec.exclude {v} should not contain any named variables"
+            )
+        return v
+
+    @cached_property
+    def is_s3(self):
+        return is_s3_uri(self.include)
+
+    @cached_property
+    def parsable_include(self):
+        parsable_include = PathSpec.get_parsable_include(self.include)
+        logger.debug(f"Setting _parsable_include: {parsable_include}")
+        return parsable_include
+
+    @cached_property
+    def compiled_include(self):
+        compiled_include = parse.compile(self.parsable_include)
+        logger.debug(f"Setting _compiled_include: {compiled_include}")
+        return compiled_include
+
+    @cached_property
+    def glob_include(self):
+        glob_include = re.sub(r"\{[^}]+\}", "*", self.include)
+        logger.debug(f"Setting _glob_include: {glob_include}")
+        return glob_include
+
+    @pydantic.root_validator()
+    def validate_path_spec(cls, values: Dict) -> Dict[str, Any]:
+
+        # validate that main fields are populated
+        required_fields = ["include", "file_types", "default_extension"]
+        for f in required_fields:
+            if f not in values:
+                logger.debug(
+                    f"Failed to validate because {f} wasn't populated correctly"
```

Comment:
do we want to throw an error here?

Reply:
throwing an exception here obscures the main failure that might have been already caught e.g. by the validators that are attached to the "include" field. 
That is the main reason why I added this here.